packages/flutter_tools/test/asset_bundle_package_test.dart

```diff
@@ -447,4 +457,244 @@ $assetsSection
+        assetsOnDisk,
+        <String>['test_package'],
+        expectedAssetManifest,
+      );
+    });
+
+    testUsingContext(
+        'One asset is bundled with variant, scanning wrong directory', () async {
+      establishFlutterRoot();
+
+      writePubspecFile('pubspec.yaml', 'test');
+      writePackagesFile('test_package:p/p/lib/');
+
+      final List<String> assetsOnDisk = <String>['a/foo','a/b/foo','a/bar'];
+      final List<String> assetOnManifest = <String>['a','a/bar']; // can't list 'a' as asset, should be 'a/'
+
+      writePubspecFile(
+        'p/p/pubspec.yaml',
+        'test_package',
+        assets: assetOnManifest,
+      );
+
+      writeAssets('p/p/', assetsOnDisk);
+
+      final AssetBundle bundle = AssetBundleFactory.instance.createBundle();
+      await bundle.build(manifestPath: 'pubspec.yaml');
+      assert(bundle.entries['AssetManifest.json'] == null,'Invalid pubspec.yaml should not generate AssetManifest.json'  );
+    });
+  });
+
+  group('AssetBundle assets from scanned paths with MemoryFileSystem', () {
+    String readSchemaPath(FileSystem fs) {
+      final String schemaPath = buildSchemaPath(fs);
+      final File schemaFile = fs.file(schemaPath);
+
+      return schemaFile.readAsStringSync();
+    }
+
+    void writeSchema(String schema, FileSystem filesystem) {
+      final String schemaPath = buildSchemaPath(filesystem);
+      final File schemaFile = filesystem.file(schemaPath);
+
+
+      final Directory schemaDir = filesystem.directory(
+          buildSchemaDir(filesystem));
+
+      schemaDir.createSync(recursive: true);
+      schemaFile.writeAsStringSync(schema);
+    }
+    //////////////////////////////////////
+    void testUsingContextAndFs(String description, dynamic testMethod(),) {
+      final FileSystem windowsFs = new MemoryFileSystem(
+          style: FileSystemStyle.windows);
+
+      final FileSystem posixFs = new MemoryFileSystem(
+          style: FileSystemStyle.posix);
+
+      const String _kFlutterRoot = '/flutter/flutter';
+      establishFlutterRoot();
+
+      final String schema = readSchemaPath(fs);
+
+      testUsingContext('$description - on windows FS', () async {
+        establishFlutterRoot();
+        writeSchema(schema, windowsFs);
+        await testMethod();
+      },
+          overrides: <Type, Generator>{
+            FileSystem: () => windowsFs,
+            Platform: () =>
+            new FakePlatform(environment: <String, String>{
+              'FLUTTER_ROOT': _kFlutterRoot,
+            },
+                operatingSystem: 'windows')
+          });
+
+      testUsingContext('$description - on posix FS', () async {
+        establishFlutterRoot();
+        writeSchema(schema, posixFs);
+        await testMethod();
+      },
+          overrides: <Type, Generator>{
+            FileSystem: () => posixFs,
+            Platform: () =>
+            new FakePlatform(environment: <String, String>{
+              'FLUTTER_ROOT': _kFlutterRoot,
```

Comment:
Just put the environment map on one line to make the indentation easier to read.